logging/DetailedErrorFormatter.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+namespace SilverStripe\Framework\Logging;
+
+use Monolog\Logger;
+use Monolog\Handler\AbstractProcessingHandler;
+use Monolog\Formatter\FormatterInterface;
+
+/**
+ * Monolog-compatible error handler that will output a detailed error message to the screen.
+ */
+class DetailedErrorFormatter implements FormatterInterface
+{
+	public function format(array $record)
+	{
+		if(isset($record['context']['exception'])) {
+			$exception = $record['context']['exception'];
+			$context = array(
+				'code' => $exception->getCode(),
+				'message' => 'Uncaught ' . get_class($exception) . ': ' . $exception->getMessage(),
+				'file' => $exception->getFile(),
+				'line' => $exception->getLine(),
+				'trace' => $exception->getTrace(),
+			);
+		} else {
+			$context = $record['context'];
+			foreach(array('code','message','file','line') as $key) {
+				if(!isset($context[$key])) {
+					$context[$key] = isset($record[$key]) ? $record[$key] : null;
+				}
+			}
+
+			$trace = debug_backtrace();
```

Comment:
Spoke to @tractorcow, we decided that in these cases it's probably better to avoid the trace entirely: https://gist.github.com/124f8f3d3bc8db8fbbf7